packetio/buffer.go

```diff
@@ -10,11 +10,34 @@ import (
 	"github.com/pion/transport/deadline"
 )
 
+// BufferPacketType allow the buffer to know which packet protocol is writing.
+type BufferPacketType int
+
+const (
+	// RTPBufferPacket indicates the buffer that is handling RTP packets
+	RTPBufferPacket BufferPacketType = 1
+	// RTCPBufferPacket indicates the buffer that is handling RTCP packets
+	RTCPBufferPacket BufferPacketType = 2
+)
+
 var errPacketTooBig = errors.New("packet too big")
 
-// Buffer allows writing packets to an intermediate buffer, which can then be read form.
+// Buffer is an interface to implement a writing/reading packet buffer.
+type Buffer interface {
+	Write(packet []byte) (int, error)
+	Read(packet []byte) (int, error)
+	Close() (err error)
+	Count() int
+	SetLimitCount(limit int)
+	Size() int
+	SetLimitSize(limit int)
+	SetReadDeadline(t time.Time) error
+	SetBufferMetadata(t BufferPacketType, ssrc uint32)
```

Comment:
Do we need the metadata in the base class? We may also be handling ICE and DTLS traffic.

Reply:
I could not find anothwer way to inform the buffer which SSRC and packet type is handling

Reply:
@Sean-Der  You can see what I mean in https://github.com/pion/srtp/pull/126/files